pkg/status/workstatus.go

```diff
@@ -168,3 +172,27 @@ func runtimeObjectToWorkStatusRef(obj runtime.Object) (*workStatusRef, error) {
 		SourceObjectIdentifier: objIdentifier,
 	}, nil
 }
+
+func getObjectStatusLastUpdateTime(metaObj metav1.Object) *metav1.Time {
+	// iterate through all managedFields entries to find the one that updated the status
+	latestTime := &metav1.Time{}
+
+	for _, field := range metaObj.GetManagedFields() {
+		if field.FieldsType == "FieldsV1" &&
+			field.FieldsV1 != nil {
+			// parse the FieldsV1.Raw to a map to check if "f:status" is present
+			var fieldsMap map[string]interface{}
+			if err := json.Unmarshal(field.FieldsV1.Raw, &fieldsMap); err != nil {
+				continue
+			}
+
+			if _, ok := fieldsMap["f:status"]; ok {
+				if field.Time.After(latestTime.Time) {
```

Comment:
What if `latestTime == nil`?

Reply:
It cannot be nil. 

`latestTime := &metav1.Time{}` initializes it to zero time (and grabs the pointer to it).
Afterwards, it is only updated through this line.

Reply:
Indeed, that is true of the current code. I was wondering if you had intended to return `nil` in the case of zero matching fields. If the intent is never to return a nil then why is `workStatus` declared to hold a _pointer to_ a time rather than simply a time?